### ISSUE TITLE
2.1.8 : Country picker shows 2 blank values and separator on shipping method page

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#10885]
I have a store configured with one allowed country, but on the shipping method page it shows 2 blank values and a separator:

### Preconditions

1. Magento 2.1.8

### Steps to reproduce

1. Allow only one country (in my case, this is Belgium)
2. Select a country on the shipping detail page.

### Expected result

 I should only see 1 country in the dropdown

### Actual result

I get 4 values: 2 blank lines, the country, and a separator.

<img width="494" alt="screen shot 2017-09-14 at 2 14 07 pm" src="https://user-images.githubusercontent.com/346681/30430106-62f2b61c-995a-11e7-9695-6c25c8c29a84.png">

When you view the source, you see that the blank value and separator are explicitly set:

<img width="781" alt="screen shot 2017-09-14 at 2 40 45 pm" src="https://user-images.githubusercontent.com/346681/30430200-c34cd358-995a-11e7-9a8b-5a6d6dcc99f7.png">

However, the first empty value is not, so it could be a combination of serverside/clientside.

Everything worked fine on 2.1.7. 

#10886 Could also be related to this.

**Comment by kanduvisla** [comment]
Further investigation shows me that this happens with a certain combination of configuration. When I set default country to BE, allowed countries to only BE and top destinations also to only BE this happens.

This is because of a combination of methods in `Magento\Checkout\Block\Checkout\DirectoryDataProcessor`. First `Magento\Directory\Model\ResourceModel\Country\Collection::toOptionArray()` is called in this class, which will return something like this:

```
array (
    0 =>
        array (
            'value' => '',
            'label' => ' ',
        ),
    1 =>
        array (
            'value' => 'BE',
            'label' => 'België',
            'is_region_visible' => false,
        ),
);
```
Secondly, `\Magento\Checkout\Block\Checkout\DirectoryDataProcessor::orderCountryOptions()` is going to manipulate this output according to the output of `\Magento\Directory\Helper\Data::getTopCountryCodes()`. If all of these return just one country code, the resulting output will be:

```
array (
    0 =>
        array (
            'value' => '',
            'label' => ' ',
        ),
    1 =>
        array (
            'value' => 'BE',
            'label' => 'België',
            'is_region_visible' => false,
        ),
    2 =>
        array (
            'value' => 'delimiter',
            'label' => '──────────',
            'disabled' => true,
        ),
)
```

So there is definitely some logic handling to do server side. And to make it extra quirky, the JavaScript component adds an extra blank on top of it. I guess this as to do because `" "` !== `""`, but that's just an assumption.

For now I can work around this issue by disabling any top destination, but this still needs some attention.

**Comment by kanduvisla** [comment]
Hm, disabling all top destinations still give 2 empty options:

<img width="489" alt="screen shot 2017-09-14 at 3 10 40 pm" src="https://user-images.githubusercontent.com/346681/30431479-f586b20e-995e-11e7-8f91-15184510a985.png">

Guess I'll write a plugin for now to solve this issue...

**Comment by magento-engcom-team** [comment]
@kanduvisla, thank you for your report.
The issue is already fixed in 2.2.0

**Comment by rohit-incedo** [comment]
@magento-engcom-team how to fix this in version 2.1.9?

**Comment by erlisdhima** [comment]
@rohit-incedo I made [this patch](https://github.com/magento/magento2/files/1798537/fix-checkout-country-dropdown.txt) as a quick fix for 2.1.x.

**Comment by Ctucker9233** [comment]
@magento-engcom-team What are the commits for this issue?
